python/cudf/cudf/core/mixins/mixin_factory.py

```diff
@@ -0,0 +1,217 @@
+# Copyright (c) 2022, NVIDIA CORPORATION.
+
+import inspect
+
+
+# A simple `partialmethod` that allows setting attributes such as
+# __doc__ on instances.
+def _partialmethod(method, *args1, **kwargs1):
```

Comment:
is this at all like `functools.partialmethod`?

Reply:
Yes -- except we run into exactly the issue described in [this](https://stackoverflow.com/a/32997046/1320147) Stack Overflow question, so we need this hack.

Reply:
I tried to explain it better.